cmd/flagsaver/flagsaver.go

```diff
@@ -0,0 +1,36 @@
+// Package flagsaver provides a simple way to save and restore flag values.
+// TODO(RJPercival): Move this to its own GitHub project.
+//
+// Example:
+//   func TestFoo(t *testing.T) {
+//     defer flagsaver.Save().Restore()
+//     // Test code that changes flags
+//   } // flags are reset to their original values here.
+package flagsaver
+
+import "flag"
+
+// Stash holds flag values so that they can be restored at the end of a test.
+type Stash struct {
```

Comment:
That would preclude extending this type in the future and expose the implementation detail that it stores a map of flag names to values. The only reason I can see to do this is to allow the user to examine stashed flag values - do you see that being a use case?